reconcile/cna/assets/asset.py

```diff
@@ -228,6 +228,40 @@ def from_api_mapping(
             **params,
         )
 
+    @classmethod
+    def _get_query_class_type(cls) -> Type[AssetQueryClass]:
+        return get_args(cls.__orig_bases__[0])[0]  # type: ignore[attr-defined]
+
+    @classmethod
+    def _get_config_class_type(cls) -> Type[ConfigClass]:
+        return get_args(cls.__orig_bases__[0])[1]  # type: ignore[attr-defined]
+
+    @classmethod
+    def _get_config(
+        cls, attribute: str, spec: AssetQueryClass
+    ) -> Optional[ConfigClass]:
+        if not (configs := getattr(spec, attribute, None)):
+            return None
+        config_class = cls._get_config_class_type()
+        if isinstance(configs, config_class):
+            return configs
+        else:
+            raise AssetError(
+                f"{attribute} for asset {spec.provider}:{spec.identifier} are not of expected type {config_class}"
+            )
+
+    @classmethod
+    def aggregate_config(cls, spec: AssetQueryClass) -> ConfigClass:
+        defaults = cls._get_config("defaults", spec)
+        overrides = cls._get_config("overrides", spec)
+        config_class = cls._get_config_class_type()
+        data = {
+            property: getattr(overrides, property, None)
+            or getattr(defaults, property, None)
+            for property in config_class.__annotations__.keys()
```

Comment:
Can a defaults file have nested parameters? Will this work properly on a nested defaults section?

Reply:
Lets consider this

**defaults**
```yaml
param_a:
  attr_a: something
  attr_b: else
```

**overrides**
```yaml
param_a:
  attr_a: different
```

This will remove `attr_b` from the final map, even though intention was to only override `attr_a`.
Not sure if this is a valid case or if we assume that we always override all subkeys.

Reply:
you are right. it won't work in those scenarios. e.g. 

```yaml
defaults:
  parent_field:
    nested_field_1: default_1
    nested_field_1: default_2
overrides:
  parent_field:
    nested_field_2: override_2
```

would result in 

```yaml
parent_field:
    nested_field_2: override_2
```

instead of 

```yaml
parent_field:
    nested_field_1: default_1
    nested_field_2: override_2
```

i can get that resolved properly in a followup PR if you want

Reply:
If this is intended then its fine. Im not even sure this is a proper case to be concerned about :sweat_smile: